main.tf

```
@@ -47,16 +47,17 @@ resource "aws_subnet" "public" {
       "Name" = format(
         "%s-%s",
         var.name,
-        count.index,
+        each.key,
```

```suggestion
        each.value,
```
could we consider use value here?

---

In this case key is a region name, like `eu-central-1a`. 
In previous version numeric index was used, like `redirect-0`.
IMHO, it gives easy overview for the subnet name to have region as a part of their `Name`, like `redirect-eu-central-1a`.